### ISSUE TITLE
Move `onClick` function to the button 

### ISSUE THREAD

**Issue by ecurrencyhodler** [saving-satoshi/saving-satoshi#528]
The `onClick` function still has to be moved to the button, as I pointed out multiple times. It cannot be on the image.

When I use VoiceOver right now, I can highlight the different avatars. From a technical perspective, the button element is highlighted (not the image). But pressing does not do anything. That's because the `onClick` is not on the button. 

`key={i}` can also be removed from the Avatar, since the mapping happens on the button element.

_Originally posted by @GBKS in https://github.com/saving-satoshi/saving-satoshi/issues/499#issuecomment-1600235744_
